Packs/AnomaliThreatStreamFeed/Integrations/AnomaliThreatStreamFeed/AnomaliThreatStreamFeed.py

```diff
@@ -0,0 +1,544 @@
+            else:
+                url_suffix_for_next_page = next_page  # Use as is if it's already relative
+
+            demisto.debug(f"{THREAT_STREAM} - Fetching next page: {url_suffix_for_next_page}")
+            response = client.http_request(method="GET", url_suffix=url_suffix_for_next_page)
+            current_page_indicators = response.get("objects", [])
+
+            if current_page_indicators:
+                all_raw_indicators.extend(current_page_indicators)
+            else:
+                demisto.debug(f"{THREAT_STREAM} - No more indicators found on current page during pagination, breaking.")
+                break
+
+            next_page = response.get("meta", {}).get("next", None)
+        except Exception as e:
+            demisto.error(f"{THREAT_STREAM} - Error during pagination: {e}. Continuing with fetched indicators.")
+            break  # Break pagination on error but process what's already fetched
+
+    if not all_raw_indicators:
+        demisto.info(f"{THREAT_STREAM} - No new indicators found since last run or no indicators matching criteria.")
+        # Update last_run even if no indicators are found, to avoid refetching the same period.
+        return now.strftime("%Y-%m-%dT%H:%M:%SZ"), []
+
+    demisto.debug(f"{THREAT_STREAM} - Total raw indicators fetched: {len(all_raw_indicators)}")
+    parsed_indicators_list: list[dict[str, Any]] = []
+
+    for indicator_raw in all_raw_indicators:
+        try:
+            # Parse each raw indicator and calculate its DBot score
+            parsed_indicator = parse_indicator_for_fetch(indicator_raw, tlp_color, create_relationship, reliability)
+            parsed_indicators_list.append(parsed_indicator)
+        except Exception as e:
+            demisto.error(f"{THREAT_STREAM} - Error parsing indicator ID {indicator_raw.get('id')}:{e}. Skipping this indicator.")
+            continue  # Continue to the next indicator even if one fails
+
+    demisto.debug(f"{THREAT_STREAM} - Successfully parsed {len(parsed_indicators_list)} indicators for fetch.")
+    # Return the current UTC timestamp for the next successful run and the list of parsed indicators
+    # todo here
+    return now.strftime("%Y-%m-%dT%H:%M:%SZ"), parsed_indicators_list
+
+
+def create_relationships(create_relationships_param: bool, reliability: str, indicator: dict[str, Any]) -> list[dict[str, Any]]:
+    """
+    Generates a list of relationship objects for a given indicator.
+    Handles single and multiple related entities.
+
+    Args:
+        create_relationships_param (bool): Whether to enable relationship creation.
+        reliability (str): The reliability score for the relationship source.
+        indicator (dict[str, Any]): The raw indicator dictionary.
+
+    Returns:
+        list[dict[str, Any]]: A list of relationship dictionaries formatted for Cortex XSOAR.
+    """
+    relationships: list[dict[str, Any]] = []
+
+    if not create_relationships_param:
```

Comment:
A suggestion - maybe not calling the function at all if create_relationships_param=False (then we don't need this arg here).

Reply:
@Shir2611 - WDYT?